GOVERNANCE.md

```diff
@@ -0,0 +1,62 @@
+# Project Governance
+
+The Valkey project is managed by a Technical Steering Committee (TSC) composed of the maintainers of Valkey repository.
+The Valkey project includes all of the repositories under the Valkey-io organization.
+Maintainers are defined as individuals with full commit access to the Valkey repository as well as additional members defined in the [MAINTAINERS.md](MAINTAINERS.md) file.
+Maintainers of other repositories within the Valkey project are not members of the TSC unless explicitly added.
+
+## Technical Steering Committee
+
+The TSC will be responsible for oversight of all technical, project, approval, and policy matters for Valkey.
+
+The TSC members are listed in the [MAINTAINERS.md](MAINTAINERS.md) file in the Valkey repository.
+New maintainers (and accordingly, TSC members) may be added or removed by no less than 2/3 affirmative vote of the current TSC.
+The TSC will appoint a Chair responsible for organizing TSC meetings.
+If the TSC Chair is removed from the TSC (or the Chair steps down from that role), it is the responsibility of the TSC to appoint a new Chair.
+The TSC can amend this governance document by no less than a 2/3 affirmative vote.
+
+The TSC may, at its discretion, add or remove members who are not maintainers of the main Valkey repository.
+The TSC may, at its discretion, add or remove maintainers from other repositories within the Valkey project.
+
+## Voting
+
+The TSC will strive for all decisions to be made by consensus.
+While explicit agreement of the entire TSC is preferred, it is not required for consensus.
+Rather, the TSC will determine consensus based on their good faith consideration of a number of factors, including the dominant view of the TSC and nature of support and objections.
+The TSC will document evidence of consensus in accordance with these requirements.
+If consensus cannot be reached, the TSC will make the decision by a vote.
+
+A vote will also be called when an issue or pull request is marked as a major decision, which are decisions that have significant impact on the Valkey architecture or design.
+Examples of major decisions:
+    * Fundamental changes to the Valkey core datastructures
+    * Adding a new data structure or API
+    * Changes that affect backward compatibility
+    * New user visible fields that need to be maintained
+    * Modifications to the TSC or other governance documents
+
+The TSC Chair will call a vote with reasonable notice to the TSC, setting out a discussion period and a separate voting period.
+Any discussion may be conducted in person or electronically by text, voice, or video.
+The discussion will be open to the public, with the notable exception of discussions involving embargoed security issues or the addition or removal of maintainers, which will be private.
```

Comment:
>The discussion will be open to the public

Should we talk about the form of "public disclosure" of these major decision discussions in this doc? or it will be clarified separately?

Reply:
I tried to be vague when I can be. Do you think we need more detail on what counts as public?

Reply:
maybe not in this doc but would be good to document it somewhere else. not a blocker for this PR.

Reply:
I put this in notes for next meeting.